### ISSUE TITLE
Double vision when game doesn't reach target framerate

### ISSUE THREAD

**Issue by OrionMoonclaw** [WiVRn/WiVRn#142]
I've noticed there's some double vision when the game fails to meet the target framerate and there's a bunch of frame late by 8.33ms lines in the log. This is only visible when I'm moving my head, it doesn't appear when looking around or moving my hand, it's very consistent (likely every other frame is wrong) and makes the image look fuzzy unless you look at a thin object where you can see it's getting doubled up.

Should hopefully be reproducible by launching any laggy game (Resonite, VRC) at a high enough resolution to not reach the target

**Comment by galister** [comment]
Care to share your GPU model and driver version? How does `nvtop` look while this is happening?

Also, did you `setcap CAP_SYS_NICE=eip` the `wivrn-server` binary?

**Comment by OrionMoonclaw** [comment]
Both that and the VR power profile are set

[WiVRn v0 19 1 87 g66aa729.txt](https://github.com/user-attachments/files/17318985/WiVRn.v0.19.1.87.g66aa729.txt)
That's the full WiVRn log

GPU is RX 6800, with Mesa 24.2.4 on Arch

This is how it looks like when the game keeps up, no double vision except for when there are visible spikes on the graph:
![JPEG_20241009_234608_1029465517653524524](https://github.com/user-attachments/assets/4f4f85a1-3993-4ec3-a1bf-32fcca30b4db)
![Screenshot From 2024-10-09 23-38-25](https://github.com/user-attachments/assets/647df805-b04d-4a90-a46f-44c2989b5500)

And this when it has to drop down to 60 or lower and there is constant double vision:
![JPEG_20241009_234556_3851912982763052220](https://github.com/user-attachments/assets/66767b8b-0cbc-4ad7-bbed-6691acb9efcd)
![Screenshot From 2024-10-09 23-37-53](https://github.com/user-attachments/assets/c0a54816-42c7-496e-bfd0-daf0299101bb)

One thing I sadly didn't catch on the screenshot was a tight zigzag pattern that often appears on the graph with the predicted and displayed timings

**Comment by OrionMoonclaw** [comment]
I was able to capture this on video somewhat, this is me turning myself with the controller and cancelling it out by turning my head the opposite direction, you can see the view shake. In person the two views swap so fast it just feels like double vision. Resonite is running at 40~ fps here, with WiVRn set to 120.

https://github.com/user-attachments/assets/6760a935-e24d-47b7-b2a7-de7efb0ef48b

**Comment by galister** [comment]
looking at the cpu + gpu times, i think the headset might be struggling to hit 8.33 ms and that's why you're seeing the stutters. does it also happen at 90hz?

**Comment by OrionMoonclaw** [comment]
Yep, at 90 Hz it becomes much more jarring since I can now make out the individual frames when it happens

**Comment by OrionMoonclaw** [comment]
Sadly not, what would be the best information to collect here?

Note that this doesn't affect normally looking around, that works fine regardless of framerate, it's only in-game movements that cause double vision. If I rotate in-game with my controller and move my head the other way (so the motion is supposed to cancel out), that's when I see double, if I just turn my head it's all fine. The same thing happens if I turn in-game and track an object with my eyes. It mostly makes things difficult to read when I'm in motion.

**Comment by galister** [comment]
Hi, considering it's been some time, mind sharing how the experience is with recent WiVRn builds?

**Comment by galister** [comment]
Closing due to no response. If you have fresh info, please create a follow-up ticket.
